### PR TITLE
Fix circular imports

### PR DESCRIPTION
**Pull request by nikmo33**
Before this a simple script in the scripts folder that imports a DataParser failed with the following error - 

```
Python 3.8.13 (default, Oct 21 2022, 23:50:54) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from nerfstudio.data.dataparsers.base_dataparser import DataparserOutputs
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/nikhil/nerfstudio-main/nerfstudio/data/dataparsers/base_dataparser.py", line 28, in <module>
    import nerfstudio.configs.base_config as cfg
  File "/home/nikhil/nerfstudio-main/nerfstudio/configs/base_config.py", line 193, in <module>
    from nerfstudio.pipelines.base_pipeline import VanillaPipelineConfig
  File "/home/nikhil/nerfstudio-main/nerfstudio/pipelines/base_pipeline.py", line 41, in <module>
    from nerfstudio.data.datamanagers.base_datamanager import (
  File "/home/nikhil/nerfstudio-main/nerfstudio/data/datamanagers/base_datamanager.py", line 38, in <module>
    from nerfstudio.data.dataparsers.base_dataparser import DataparserOutputs
ImportError: cannot import name 'DataparserOutputs' from partially initialized module 'nerfstudio.data.dataparsers.base_dataparser' (most likely due to a circular import) (/home/nikhil/nerfstudio-main/nerfstudio/data/dataparsers/base_dataparser.py)
```
After
```

Python 3.8.13 (default, Oct 21 2022, 23:50:54) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from nerfstudio.data.dataparsers.base_dataparser import DataparserOutputs
>>> 
```

### CODE REVIEW

File: nerfstudio/configs/config.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+from nerfstudio.configs.config_utils import to_immutable_dict
+from nerfstudio.engine.optimizers import OptimizerConfig
+from nerfstudio.engine.schedulers import SchedulerConfig
+from nerfstudio.pipelines.base_pipeline import VanillaPipelineConfig
+
+CONSOLE = Console(width=120)
+
+
+@dataclass
+class Config(PrintableConfig):
```
Comment (tancik) [inline nerfstudio/configs/config.py:42]: Maybe rename file and config `ExperimentConfig`

**Review by tancik — APPROVED** [review]
Thanks for digging into this! These config imports have caused quite a bit of confusion.

File: nerfstudio/configs/config.py (hunk 1 above)
Comment (nikmo33) [reply nerfstudio/configs/config.py:42]: that what I was thinking too..

**Comment by nikmo33** [comment]
> Not sure what's up with the rtd build, it can't pull the code image I don't think it is related to the code. I can force merge if you think the rest of the code is ready to merge.

Yup the rest of the code is done. Happy to merge when you're ready

**Pull request merged**